module/executiondatasync/pruner/pruner.go

```diff
@@ -0,0 +1,206 @@
+package pruner
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/executiondatasync/tracker"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/util"
+)
+
+const (
+	defaultHeightRangeTarget = uint64(400_000)
+	defaultThreshold         = uint64(100_000)
```

Comment:
These seem to be very large number? Any reasoning why we pick them as is?

Especially the default threshold, this would cause each time to prune 100_000 height, which is a very big number of keys to remove.

I'm thinking about `100`.

Reply:
We can use a different number for threshold. The smaller it is, the more frequent the pruning. if we pick 100, that would mean pruning happens approximately every 2 minutes, which seems quite frequent